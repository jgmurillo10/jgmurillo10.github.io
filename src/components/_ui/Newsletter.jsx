import React from "react"
import styled from "@emotion/styled"
import MailchimpSubscribe from "react-mailchimp-subscribe"

const Subscribe = styled("div")`
  box-shadow: 0px 9px 24px rgb(0 0 0 / 6%);
  max-width: 550px;
  margin: 90px auto 0;
  padding: 24px;
  h4 {
    padding: 12px;
  }
  button {
    background-color: #73abff;
    border: none;
    color: white;
    font-size: 1rem;
    font-weight: 600;
    margin: 12px;
    outline: none;
    padding: 1em 2em;
    width: 100%;

    &:focus {
      outline: auto 5px -webkit-focus-ring-color;
    }
  }
  input {
    width: 100%;
    border: none;
    border-bottom: 1px solid #ccc;
    margin: 12px;
    font-size: 1rem;
  }

  > div {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;

    > div {
      display: block;
      width: 100%;
      margin: 12px;
    }
  }
`

const url =
  "https://juanmurillo.us20.list-manage.com/subscribe/post?u=8099ac62ba887740228d614c1&id=1ffbb498a4"

const Newsletter = ({ body }) => (
  <Subscribe>
    <h4>{body}</h4>
    <MailchimpSubscribe url={url} />
  </Subscribe>
)

export default Newsletter
