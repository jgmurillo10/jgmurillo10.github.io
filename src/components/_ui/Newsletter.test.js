import React from "react"
import renderer from "react-test-renderer"
import Newsletter from "./Newsletter"

describe("Newsletter", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Newsletter body="Body" />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
