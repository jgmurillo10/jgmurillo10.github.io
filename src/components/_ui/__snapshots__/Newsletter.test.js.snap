// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-r6e1pm{box-shadow:0px 9px 24px rgb(0 0 0 / 6%);margin:90px auto 0;max-width:550px;padding:24px;}.css-r6e1pm h4{padding:12px;}.css-r6e1pm button{background-color:#73abff;border:none;color:white;font-size:1rem;font-weight:600;margin:12px;outline:none;padding:1em 2em;width:100%;}.css-r6e1pm button:focus{outline:auto 5px -webkit-focus-ring-color;}.css-r6e1pm input{border:none;border-bottom:1px solid #ccc;font-size:1rem;margin:12px;padding:8px;width:100%;}.css-r6e1pm >div{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex;-webkit-box-flex-wrap:wrap;-webkit-flex-wrap:wrap;-ms-flex-wrap:wrap;flex-wrap:wrap;-webkit-box-pack:justify;-webkit-justify-content:space-between;justify-content:space-between;}.css-r6e1pm >div >div{display:block;margin:12px;width:100%;}",
      }
    }
    data-emotion="css r6e1pm"
  />,
  <div
    className="css-r6e1pm"
  >
    <h4>
      Body
    </h4>
    <div>
      <input
        placeholder="Your email"
        type="email"
      />
      <button
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </div>,
]
`;
